Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.h

```diff
@@ -53,10 +50,10 @@ class RemoteDisplayListRecorderProxy : public WebCore::DisplayList::Recorder {
     template<typename T>
     void send(T&& message)
     {
-        if (UNLIKELY(!m_renderingBackend))
+        if (UNLIKELY(!(m_renderingBackend && m_imageBuffer)))
```

Comment:
Rather than bury this here in `send()`, can we piggy-back off `appendStateChangeItemIfNecessary()`, adding some logic in `DisplayList::Recorder` that knows when a drawing command was submitted?

Reply:
A new virtual method `setNeedFlush()` is added to `DisplayList::Recorder` and it is called from its `appendStateChangeItemIfNecessary()`. `RemoteDisplayListRecorderProxy` overrides this function to call `ImageBuffer::setNeedsFlush()`.